### ISSUE TITLE
ReadOnly persistent volume not respected

### ISSUE THREAD

**Issue by sahewat** [polyaxon/polyaxon#199]
When using an attached persistent volume with the read-only flag an error is thrown during an experiment run and the sidecar shows that said volume is mounted with `read-only: false`
## Helm Configuration (snippet):
```
persistence: 
  data:
    data1:
      existingClaim: polyaxon-pvc-persistent-hazard
      mountPath: /persistent-hazard
      readOnly: true
```
## Experiment Permission:
```
Volumes:
  data1:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  polyaxon-pvc-persistent-hazard
    ReadOnly:   false
```

**Comment by mouradmourafiq** [comment]
@sahewat Can you explain a bit what sidecar you are having issues with?
AFAIK Polyaxon sidecar does not mount the users' volumes, the volumes are only mounted for experiments' and jobs' pods.

**Comment by sahewat** [comment]
Thank you @mouradmourafiq! I may have been mistaken about the sidecar (I thought I saw the error there as well). This definitely happens with both notebooks and experiments though.
(I'll change the issue to reflect this)

**Comment by mouradmourafiq** [comment]
Just created a volume and used it to start an experiment with volume in read-only mode, the volume was mounted correctly to the experiment:

```
            "name": "data2",
            "readOnly": true,
            "mountPath": "/data2"
```

I also started a notebook, and the volume was also mounted correctly:

```
            "name": "data2",
            "readOnly": true,
            "mountPath": "/data2"
```

I need more information to be able to reproduce the issue.

**Comment by mouradmourafiq** [comment]
Also I am not sure I understand what do you mean by "experiment permission". Is it the definition you get when you look at the experiment pod? or is some section you added to the polyaxonfile for your experiment?

**Comment by mouradmourafiq** [comment]
I just created a new pvc with `ReadOnlyMany` (but no with `gcePersistentDisk`), and I started an experiment, the readOnly is still respected.

`Specifically, I'm getting a FailedMount error when multiple pods try to access the data.` Do you get this error when creating only one experiment at a time, or when you start multiple ones?

**Comment by sahewat** [comment]
Only when using multiple experiments. It also happens when another instance (not polyaxon) has the volume mounted in readOnly (which is expected)

**Comment by mouradmourafiq** [comment]
I assume that this is not a Polyaxon related issue, but instead it's an issue/misconfiguration of the pvc, I also assume that, independently of Polyaxon, you won't probably be able to mount that volume to many pods at the same time (you can try that and see, it's also easier to debug these type of issues when playing with a busybox pod).

I would try to ask the question to a larger community, stackoverflow or the k8s community for instance.

I am not sure but it could be an [access control mounting issue](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/#access-control), I remeber having similar issues in the past with a volume backed with azure files, you can look at this definition from the [polyaxon examples](https://github.com/polyaxon/polyaxon-examples/blob/master/azure/data-pvc.yml#L5)

**Comment by sahewat** [comment]
I actually am able to mount it in multiple machines simultaneously outside of Polyaxon. I've tried this with both VMs and Kubeflow.

**Comment by mouradmourafiq** [comment]
Ok, this is strange, so you are able to mount the volume to 2 pods running simultaneously outside of Polyaxon?

Can you please run 2 experiments on Polyaxon, and put the part of the spec here for the volumes and volume mounts.

**Comment by mouradmourafiq** [comment]
I think I see where the problem is, Polyaxon is only specifying the `readOnly` on mounts but not on volumes.

**Comment by sahewat** [comment]
Interesting. Is it an easy fix?

**Comment by mouradmourafiq** [comment]
The fix will be part of the next release (v0.2.7), the release is planned for Monday.

**Comment by mouradmourafiq** [comment]
Feel free to reopen this if it's still not working after upgrading to 0.2.7.
